### PR TITLE
Fix segfaults

### PR DESCRIPTION
**Pull request by msantos**
Fix some crashes found by running scan-build and AFL.

### CODE REVIEW

**Comment by yrutschle** [comment]
Thanks -- some of these I've been meaning to add for a while, and some I never thought of.

Can I just request you factorise all the:
```
ptr = malloc();
if (!ptr) { perror(); exit(); }
```

into something like:
```
ptr = malloc();
CHECK_ALLOC(ptr, "malloc");
```

with CHECK_ALLOC() to be added in common.h:
```
#define CHECK_ALLOC(a, str) \
    if (!a) { \
        fprintf(stderr, "%s:%d:", __FILE__, __LINE__); \
        perror(str); \
        exit(1); \
    }
```

Same idea as the CHECK_RES_DIE() macro: avoid template code creep if possible.
Cheers,
Y.

**Comment by msantos** [comment]
@yrutschle done, thanks for the feedback!

File: common.c [hunk 1]
```diff
@@ -84,6 +84,7 @@ int get_fd_sockets(int *sockfd[])
     if (sd > 0) {
       int i;
       *sockfd = malloc(sd * sizeof(*sockfd[0]));
+      CHECK_ALLOC(sockfd, "malloc");
```
Comment (yrutschle) [inline common.c]: *sockfd, rather

File: common.c [hunk 2]
```diff
@@ -116,10 +117,16 @@ int start_listen_sockets(int *sockfd[], struct addrinfo *addr_list)
...
+   if (num_addr == 0) {
+       fprintf(stderr, "FATAL: No available addresses.\n");
+       exit(1);
+   }
+
    if (verbose)
        fprintf(stderr, "listening to %d addresses\n", num_addr);
 
    *sockfd = malloc(num_addr * sizeof(*sockfd[0]));
+   CHECK_ALLOC(sockfd, "malloc");
```
Comment (yrutschle) [inline common.c]: CHECK_ALLOC(*sockfd, ...)

**Comment by msantos** [comment]
Fixed, thanks!

**Comment by yrutschle** [comment]
Thank you for your contribution!

**Pull request merged**